objectivec/GPBUtilities.m

```diff
@@ -1137,8 +1316,40 @@ void GPBSetMessageRepeatedField(GPBMessage *self, GPBFieldDescriptor *field, id
   GPBSetObjectIvarWithField(self, field, array);
 }
 
-#if defined(DEBUG) && DEBUG
-static NSString *TypeToStr(GPBDataType dataType) {
+BOOL GPBDataTypesEquivalent(GPBDataType type1, GPBDataType type2) {
+  return GPBBaseDataType(type1) == GPBBaseDataType(type2);
+}
+
+GPBDataType GPBBaseDataType(GPBDataType type) {
+  switch(type) {
+    case GPBDataTypeSFixed32:
+    case GPBDataTypeInt32:
+    case GPBDataTypeSInt32:
+    case GPBDataTypeEnum:
```

Comment:
I guess mapping enum here is ok with the current uses, but I wonder if it should be ensure it is enum?

Reply:
I wasn't that concerned about it. Want me to fix it up?

Reply:
I guess we should be safe with current uses, with the naming helping indicate it is internal, it likely isn't as much of an issue.